checkport_test.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"net"
+	"strconv"
+	"testing"
+)
+
+func TestCheckPortAvailability(t *testing.T) {
+	tests := []struct {
+		port int
+	}{
+		{9000},
+		{10000},
+	}
+	for _, test := range tests {
+		// This test should pass if the ports are available
+		err := checkPortAvailability(test.port)
+		if err != nil {
+			t.Fatalf("checkPortAvailability test failed for port: %d. Error: %s", test.port, err)
+		}
+
+		// Now use the ports and check again
+		ln, err := net.Listen("tcp", net.JoinHostPort("127.0.0.1", strconv.Itoa(test.port)))
```

Comment:
The best way to generate a problem  on windows would be to listen on 

```
net.Listen("tcp", fmt.Sprintf(":%d", test.port))
```

That way it listens on all interfaces. There are other problems though over this verification 

https://github.com/golang/go/issues/7598